### ISSUE TITLE
Reconstructed images are not like the input images

### ISSUE THREAD

**Issue by swyoon** [chaiyujin/glow-pytorch#20]
Hi,

First of all, thank you for a nice repo.

I am trying to map an image `x` to a latent representation `z` and then map back to its reconstruction `x_hat` with a trained model, but `x` and `x_hat` are not similar at all.

I understand that it may not be totally identical due to `Split2d` layer, but the degree of difference is way too severe.

I ran the training script, and the reconstructed images shown in Tensorboard are much similar to their corresponding inputs.

Here are rough snippets that can reproduce my problem.

For defining and loading dataset and model,
```python
from torchvision import transforms
from glow.config import JsonConfig
from glow.builder import build
from glow.trainer import Trainer
from glow.utils import load
import vision

hparams = JsonConfig('hparams/celeba.json')

transform = transforms.Compose([
    transforms.CenterCrop(hparams.Data.center_crop),
    transforms.Resize(hparams.Data.resize),
    transforms.ToTensor()])
dataset = vision.Datasets['celeba']
dataset = dataset('<path-to-CelebA>', transform=transform)

built = build(hparams, True)
load('glow_celeba.ckpt.pkg', built['graph'])
graph = built['graph']
graph. eval()
```

Now I want to encode a batch of images and decode them back.

```python
img_x = torch.stack([dataset[i]['x'] for i in range(12)])
img_x = img_x.cuda()

# encode
z, nll, y_logits = graph(img_x, y_onehot=None)

# decode
x_hat = graph(z, reverse=True)
```

When I visualize `img_x` and `x_hat`, I found a significant discrepancy.

```python
import matplotlib.pyplot as plt

# original images
np_img_x = img_x.permute(0,2,3,1).detach().cpu().numpy()
fig, axs = plt.subplots(ncols=5, figsize=(15,5))
for i in range(5):
    axs[i].imshow(np_img_x[i])
```

![image](https://user-images.githubusercontent.com/10072033/72403331-0f071a80-3795-11ea-89b6-3beff1d7781f.png)

```python
# reconstructed images
np_x_hat = x_hat.permute(0,2,3,1).detach().cpu().numpy()
fig, axs = plt.subplots(ncols=5, figsize=(15,5))
for i in range(5):
    axs[i].imshow(np_x_hat[i])
```

![image](https://user-images.githubusercontent.com/10072033/72403339-162e2880-3795-11ea-86f0-8067cb28052b.png)

As you can see, those images are very different.

As a sanity check, I ran unconditioned sampling. It gives reasonably fine images, especially with apt choice `eps_std`, so the model is well-trained.

```python
out = graph(z=None, reverse=True, eps_std=0.6)
np_out = out.permute(0,2,3,1).detach().cpu().numpy()
fig, axs = plt.subplots(ncols=5, figsize=(15,5))
for i in range(5):
    axs[i].imshow(np_out[i])
```

![image](https://user-images.githubusercontent.com/10072033/72403517-a7050400-3795-11ea-8e87-a3fc4706056f.png)

**Comment by fllrvr** [comment]
Thank you for your nice repo!

As far as I understand the multi-scale structure, `z2` defined in `Split2d` (in the non-reverse mode) should be considered as a part of the output of the normalizing flow (but should not be carried to the next `SqueezingLayer` ).
Thus when we synthesize an image, we must use a noise array whose "size" is larger than that of the current `z`.  
https://github.com/chaiyujin/glow-pytorch/blob/487a6b149295f4ec4b36e408f63604c593ff2031/glow/modules.py#L299-L304

**Comment by turnip001** [comment]
For a glow model, x and z should have the same shape. However, part of z is abandoned during the split2d operation. That may be the reason why x and x_hat are not similiar at all.

**Comment by tenpercent** [comment]
```python
# decode
x_hat = graph(z, reverse=True)
```

If you leave the parameter unnamed, it assumes you supply `x` as input, leaving `z` as `None`. Try 

```python
# decode
x_hat = graph(z=z, reverse=True)
```
